src/main/java/com/google/devtools/build/lib/bazel/bzlmod/IndexRegistry.java

```diff
@@ -117,18 +118,51 @@ private static class BazelRegistryJson {
     String moduleBasePath;
   }
 
-  /** Represents fields available in {@code source.json} for each version of a module. */
+  /** Represents the type field in {@code source.json} for each version of a module. */
   private static class SourceJson {
     String type = "archive";
+  }
+
+  /** Represents fields in {@code source.json} for each archive-type version of a module. */
+  private static class ArchiveSourceJson {
+    String type = "archive";
     URL url;
     String integrity;
     String stripPrefix;
     Map<String, String> patches;
     int patchStrip;
-    String path;
     String archiveType;
   }
 
+  /** Represents fields in {@code source.json} for each local_path-type version of a module. */
+  private static class LocalPathSourceJson {
+    String type = "local_path";
+    String path;
+  }
+
+  /** Represents fields in {@code source.json} for each git_repository-type version of a module. */
+  private static class GitRepoSourceJson {
+    String type = "git_repository";
+    String remote;
+    String commit;
+    String shallowSince;
+    String tag;
+    String branch;
+    boolean initSubmodules;
+    boolean recursiveInitSubmodules;
+    boolean verbose;
+    String stripPrefix;
+    List<String> patches;
+    String patchTool;
+    List<String> patchArgs;
+    List<String> patchCmds;
+    List<String> patchCmdsWin;
+    String buildFile;
+    String buildFileContent;
+    String workspaceFile;
```

Comment:
workspaceFile and workspaceFileContent are probably not needed anymore.

Reply:
Removed.